Sources/ArcGISToolkit/Components/AR/TableTopSceneView.swift

```diff
@@ -0,0 +1,288 @@
+    func updatePlane(with node: SCNNode, for anchor: ARAnchor) {
+        if initialTransformationIsSet {
+            node.removeFromParentNode()
+        }
+        
+        guard let planeAnchor = anchor as? ARPlaneAnchor,
+              let plane = node.childNodes.first as? Plane
+        else { return }
+        
+        // Update extent visualization to the anchor's new bounding rectangle.
+        if let extentGeometry = plane.node.geometry as? SCNPlane {
+            extentGeometry.width = CGFloat(planeAnchor.extent.x)
+            extentGeometry.height = CGFloat(planeAnchor.extent.z)
+            plane.node.simdPosition = planeAnchor.center
+        }
+    }
+}
+
+private extension View {
+    /// Sets a closure to perform when a single tap occurs on the view.
+    /// - Parameters:
+    ///   - action: The closure to perform upon single tap.
+    ///   - screenPoint: The location of the tap in the view's coordinate space.
+    func onSingleTapGesture(perform action: @escaping (_ screenPoint: CGPoint) -> Void) -> some View {
+        if #available(iOS 16.0, *) {
+            return self.onTapGesture { screenPoint in
+                action(screenPoint)
+            }
+        } else {
+            return self.gesture(
+                DragGesture()
+                    .onEnded { screenPoint in
+                        action(screenPoint.location)
+                    }
+            )
+        }
+    }
+}
+
+/// A helper class to visualize a plane found by ARKit.
+private class Plane: SCNNode {
+    let node: SCNNode
+    
+    init?(anchor: ARPlaneAnchor) {
+        // Create a node to visualize the plane's bounding rectangle.
+        let extent = SCNPlane(width: CGFloat(anchor.extent.x), height: CGFloat(anchor.extent.z))
+        node = SCNNode(geometry: extent)
+        node.simdPosition = anchor.center
+        
+        // `SCNPlane` is vertically oriented in its local coordinate space, so
+        // rotate it to match the orientation of `ARPlaneAnchor`.
+        node.eulerAngles.x = -.pi / 2
+        
+        super.init()
+        
+        node.opacity = 0.25
+        guard let material = node.geometry?.firstMaterial else { return nil }
+        
+        material.diffuse.contents = UIColor.white
+        
+        // Add the plane node as child node so they appear in the scene.
+        addChildNode(node)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+private extension ARSwiftUIViewProxy {
+    /// Performs a hit test operation to get the transformation matrix representing the corresponding real-world point for `screenPoint`.
+    /// - Parameter screenPoint: The screen point to determine the real world transformation matrix from.
+    /// - Returns: A `TransformationMatrix` representing the real-world point corresponding to `screenPoint`.
+    func hitTest(using screenPoint: CGPoint) -> TransformationMatrix? {
```

Comment:
"at" might be more descriptive in this case than "using".

```suggestion
    func hitTest(at screenPoint: CGPoint) -> TransformationMatrix? {
```